pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationTls.java

```diff
@@ -37,6 +39,7 @@
 public class AuthenticationTls implements Authentication, EncodedAuthenticationParameterSupport {
 
     private static final long serialVersionUID = 1L;
+    private AtomicBoolean bouncyCastleLoaded = new AtomicBoolean(false);
```

Comment:
```suggestion
    private static final AtomicBoolean bouncyCastleLoaded = new AtomicBoolean(false);
```

Reply:
Use static, since there can be multiple instances of AuthTLS in a single process

Reply:
Actually, we could even just put the initialization in a static block here. eg: 

```java
static {
    Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
}
```